_episodes/03-high-dim-modeling.md

```diff
@@ -4,43 +4,121 @@ teaching: 0
-> {: .solution}
-{: .challenge}
+The techniques breakdown into two broad categories, predictors and classifiers. Predictors are used to predict a value (or set of value) given a set of inputs, for example trying to predict the cost of something given the economic conditions and the cost of raw materials or predicting a country’s GDP given its life expectancy. Classifiers try to classify data into different categories, for example deciding what characters are visible in a picture of some writing or if a message is spam or not. Some examples might include:
+* predicting a person’s weight based on their height
+* predicting commute times given traffic conditions
+* predicting house prices given stock market prices
+* classifying if an email is spam or not
+* classifying what if an image contains a person or not
+
+In this lesson, we will work with predictive models in both a low and high-dimensional data regime.
+
+## Training data 
+* The text below is borrowed form this lesson: https://carpentries-incubator.github.io/machine-learning-novice-sklearn/01-introduction/index.html)
+* 
+Many (but not all) machine learning systems “learn” by taking a series of input data and output data and using it to form a model. The maths behind the machine learning doesn’t care what the data is as long as it can represented numerically or categorised.
+
+Typically we will need to train our models with hundreds, thousands or even millions of examples before they work well enough to do any useful predictions or classifications with them.
+
+Some systems will do training as a one shot process which produces a model. Others might try to continuosly refine their training through the real use of the system and human feedback to it. For example every time you mark an email as spam or not spam you are probably contributing to further training of your spam filter’s model.
+
+## **The curse of dimensionality**
+- The phrase, attributed to Richard Bellman, was coined to express the difficulty of using brute force (a.k.a. grid search) to optimize a function with too many input variables.
+- In machine learning, the Curse of Dimensionality refers to a set of problems that arise when working with high-dimensional data.
+  - Compute Time: Generally speaking, as the number of features in a model increases, so does the amount of time required to opitmize the model
+  - Overfitting: When there are more features than training observations, overfitting is gauranteed
+  - Interpretability: Models become more difficult to interpret as the numbers of features increase
+  - Clustering: Too many dimensions causes every observation in your dataset to appear equidistant from all the others — making it difficult to form clusters on the data. If the distances are all approximately equal, then all the observations appear equally alike (as well as equally different), and no meaningful clusters can be formed.
+
+### **Bias-Variance Tradeoff (Underftting vs Overfitting)**
```

Comment:
Don't need to bold headers - typically headers are already bold format.  Same with the header above